Assets/Plugins/CountlySDK/Services/AbstractBaseService.cs

```diff
@@ -23,6 +24,102 @@ protected AbstractBaseService(CountlyConfiguration configuration, CountlyLogHelp
             _consentService = consentService;
         }
 
+        protected IDictionary<string, object> RemoveSegmentInvalidDataTypes(IDictionary<string, object> segments)
+        {
+
+            if (segments == null || segments.Count == 0) {
+                return segments;
+            }
+
+            string moduleName = GetType().Name;
+            int i = 0;
+            List<string> toRemove = new List<string>();
+            foreach (KeyValuePair<string, object> item in segments) {
+                if (++i > _configuration.MaxSegmentationValues) {
+                    toRemove.Add(item.Key);
+                    continue;
+                }
+                Type type = item.Value?.GetType();
+                bool isValidDataType = item.Value != null
+                    && (type == typeof(int)
+                    || type == typeof(bool)
+                    || type == typeof(float)
+                    || type == typeof(double)
+                    || type == typeof(string));
+
+
+                if (!isValidDataType) {
+                    toRemove.Add(item.Key);
+                    Log.Warning("[" + moduleName + "] RemoveSegmentInvalidDataTypes: In segmentation Data type '" + type + "' of item '" + item.Key + "' isn't valid.");
+                }
+            }
+
+            foreach (string k in toRemove) {
+                segments.Remove(k);
+            }
+
+            return segments;
+        }
+
+        protected string TrimKey(string k)
+        {
+            if (k.Length > _configuration.MaxKeyLength) {
+                Log.Warning("[" + GetType().Name + "] TrimKey : Max allowed key length is " + _configuration.MaxKeyLength + ". "+ k + " will be truncated.");
+                k = k.Substring(0, _configuration.MaxKeyLength);
+            }
+
+            return k;
+        }
+
+        protected string[] TrimValues(string[] values)
+        {
+            for (int i = 0; i < values.Length; ++i) {
+                if (values[i].Length > _configuration.MaxValueSize) {
+                    Log.Warning("[" + GetType().Name + "] TrimKey : Max allowed value length is " + _configuration.MaxKeyLength + ". " + values[i] + " will be truncated.");
+                    values[i] = values[i].Substring(0, _configuration.MaxValueSize);
+                }
+            }
+
+
+            return values;
+        }
+
+        protected string TrimValue(string nameOf, string v)
+        {
+            if (v.Length > _configuration.MaxValueSize) {
+                Log.Warning("[" + GetType().Name + "] TrimValue : Max allowed" + nameOf + "length is " + _configuration.MaxValueSize + ". " + v + " will be truncated.");
```

Comment:
this log statement should be reworked so that it is not expected of the "nameOf" value to include whitespace at both ends. Those should already be included in the host string here.